pynetsnmp/netsnmp.py

```diff
@@ -717,14 +717,31 @@ class Session(object):
 
     cb = None
 
-    def __init__(self, cmdLineArgs=(), **kw):
+    def __init__(self, cmdLineArgs=(), freeEtimelist=True, **kw):
         self.cmdLineArgs = cmdLineArgs
+        self.freeEtimelist = freeEtimelist
         self.kw = kw
         self.sess = None
         self.args = None
         self._data = None  # ref to _CallbackData object
         self._log = _getLogger("session")
 
+    def _snmp_send(self, session, pdu):
+        """Allows to execute free_etimelist() after each snmp_send() call
+
+        Executes lib.free_etimelist() after the each lib.snmp_send() call if
+        `freeEtimelist` attribute is set or re-calls lib.snmp_send() otherwise.
+        This allows to free all the memory used by entries in the etimelist
+        inside the net-snmp library to proceess devices with duplicated engineID
+        PS - please note, that this feature is not supported by RFC
+        """
+
+        try:
+            return lib.snmp_send(self.sess, pdu)
```

Comment:
If you are allowing the session to be passed in as an argument, you should use that instead of the instance variable.  Otherwise remove the parameter.

Reply:
I have added changes